src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -415,35 +439,59 @@ private boolean hasOperator(final CigarElement maybeCigarElement, final CigarOpe
     /**
      * Does an insertion occur immediately before the current position on the genome?
      *
-     * @return true if yes, false if no
+     * @return true if an insertion occurs immediately before the current position on the genome, false otherwise
      */
-    public boolean isAfterInsertion() { return isAfter(getBetweenPrevPosition(), CigarOperator.I); }
+    public boolean isAfterInsertion() { return isImmediatelyAfter(CigarOperator.I); }
 
     /**
      * Does an insertion occur immediately after the current position on the genome?
      *
-     * @return true if yes, false if no
+     * @return true an insertion occurs immediately after the current position on the genome, false otherwise
      */
-    public boolean isBeforeInsertion() { return isBefore(getBetweenNextPosition(), CigarOperator.I); }
+    public boolean isBeforeInsertion() {
+        return isImmediatelyBefore(CigarOperator.I);
+    }
 
     /**
      * Does a soft-clipping event occur immediately before the current position on the genome?
      *
-     * @return true if yes, false if no
+     * @return true if a soft-clipping event occurs immediately before the current position on the genome, false otherwise
      */
-    public boolean isAfterSoftClip() { return isAfter(getBetweenPrevPosition(), CigarOperator.S); }
+    public boolean isAfterSoftClip() {
+        return isImmediatelyAfter(CigarOperator.S);
+    }
 
     /**
      * Does a soft-clipping event occur immediately after the current position on the genome?
      *
-     * @return true if yes, false if no
+     * @return true if a soft-clipping event occurs immediately after the current position on the genome, false otherwise.
+     */
+    public boolean isBeforeSoftClip() {
+        return isImmediatelyBefore(CigarOperator.S);
+    }
+
+    /**
+     * @return true if a given operator event occurs immediately before the current position on the genome, false otherwise.
+     */
+    @VisibleForTesting
+    boolean isImmediatelyAfter(final CigarOperator op) {
+        Utils.nonNull(op);
+        return atStartOfCurrentCigar() && getNeighborNotOnGenomeOperator(Direction.PREV) == op;
+    }
+
+    /**
+    * @return true if a given operator event occurs immediately after the current position on the genome, false otherwise.
      */
-    public boolean isBeforeSoftClip() { return isBefore(getBetweenNextPosition(), CigarOperator.S); }
+    @VisibleForTesting
+    boolean isImmediatelyBefore(final CigarOperator op) {
+        Utils.nonNull(op);
+        return atEndOfCurrentCigar() && getNeighborNotOnGenomeOperator(Direction.NEXT) == op;
```

Comment:
`isAfter()` didn't have this `atEndOfCurrentCigar()` check -- are you sure that your new code is equivalent to the previous version?